Source/WEB_ROOT/Inventory/psQuery/psQuery.js

```diff
@@ -0,0 +1,272 @@
+                } else {
+                    name_prefix = 'CF-[' + self.table_name + ':' + dcid + ']';
+                }
+            } else {
+                frn = self.table_number + String(dcid);
+                url = url + '&frn=' + frn;
+                if (isDelete) {
+                    name_prefix = 'DR-' + frn;
+                } else {
+                    name_prefix = '[' + self.table_number + ']';
+                }
+            }
+
+            // make sure fields is an object OR fields is 'delete'
+            if ((fields !== null && typeof fields === 'object') || isDelete) {
+                
+                // map fields to key/name pairs for helper file (psQuery.html)
+                var count = 0;
+                var params = {};
+                var valuemap = ['prim'];
+                if (isGuardian) valuemap = ['autosendupdate'];
+                if (isTeachers) valuemap = ['webasmt'];
+                
+                if (isDelete) {
+                    params.key1 = 1;
+                    params.name1 = name_prefix;
+                    valuemap.push(1);
+                    
+                } else {
+                    for (k in fields) {
+                        if (fields.hasOwnProperty(k)) {
+                            count++;
+                            params['key'+count] = count;
+                            params['name'+count] = name_prefix + k;
+                            
+                            // map the value for insertion later
+                            valuemap.push(fields[k]);
+                        }
+                    }
+                }
+                
+                // the helper has a limit to the number of fields it can create/update ... edit the file to increase
+                if (count>self.maxParams) {
+                    throw "to many items in fields parameters";
+                }
+
+                // call the helper file using jQuery's .post() method, passing the key/name gpv's
+                
+                $j.post(url, params, function(retForm) {
+                    
+                    // get the returned form and serialize the names and insert the values
+                    var formFields = $j(retForm).children("input");
+                    var fieldData = {};
+                    
+                    count = 0;
+                    formFields.each(function() {
+            			fieldData[$j(this).attr("name")] = valuemap[count];
+            			count++;
+                    });
+                    if (parentStr) fieldData[parentStr] = parentID;
+                    
+                    // add gpv's to URL (z, id_name, table_name)
+                    url = url + '&table_name=' + self.table_name + '&id_name=' + self.id_name+ '&z=1';
+                    
+                    // sumbit (post) the data to the server, retValue is the ID
+                    $j.post(url, fieldData, function(retID) {

```

Comment:
I would suggest more robust variable names. Most of these are spot on and I like them. parentID, formFields, fieldData very descriptive. Some of the others like retID are tougher to determine. Even if it's something as simply as idToReturn or returningID it helps so much.